### DISCUSSION TITLE
How does @Url.Action generate the route?

### DISCUSSION THREAD

**Discussion by weirdyang** [OrchardCMS/OrchardCore#9355] [Q&A]
I've a module named `OrchardCore.CustomURL`, with a controller `CustomURLController`
this is my startup.cs
```
        public override void Configure(IApplicationBuilder builder, IEndpointRouteBuilder routes, IServiceProvider serviceProvider)
        {
            var workflowControllerName = typeof(CustomURLController).ControllerName();

            routes.MapAreaControllerRoute(
                name: "InvokeCustomURL",
                areaName: "OrchardCore.CustomURL",
                pattern: "workflows/custom/{slug}",
                defaults: new { controller = workflowControllerName, action = nameof(CustomURLController.Invoke) }
            );

            routes.MapAreaControllerRoute(
                name: "CustomURL",
                areaName: "OrchardCore.CustomURL",
                pattern: "custom/{action}",
                defaults: new { controller = workflowControllerName });
        }
```

However when I do this:
```
        var test = @Url.Action("invoke", "CustomURL");
```
I expect test to generate `OrchardCore.CustomURL/CustomURL/invoke`, but instead it generates `OrchardCore.Workflows/CustomURL/invoke`, why is this happening?

**Comment by jtkech** [comment]
What is your module project name, assembly name, and which ID(s) did you use in your manifest file?

**Reply by weirdyang** [reply]
> What is your module project name, assembly name, and which ID(s) did you use in your manifest file?

@jtkech 

```
[assembly: Module(
    Author = "myAuthor",
    Category = "Workflows",
    Description = "Allows user defined HTTP Request URLs",
    Name = "Custom URL Http Request",
    Version = "0.0.1",
    Website = "https://mysite.co",
    Dependencies = new[] { "OrchardCore.Workflows" }
)]
```

Assembly name: `OrchardCore.CustomURL.dll`

Project name: `OrchardCore.CustomURL`

**Comment by weirdyang** [comment]
I think it's because I'm using it in a workflow activity edit page, so the the `@Url.Action()` method infers the area as `OrchardCore.Workflows`,  
if I use it like `@Url.Action(action: "invoke", controller: "CustomURL",new { area = "OrchardCore.CustomURL"})`, explicitly passing my module name as the area, it works as expcted.
